### ISSUE TITLE
Hint linking errors

### ISSUE THREAD

**Issue by noughtmare** [yi-editor/yi#870]
`:yi errorEditor "hello"` gives me: 

```
 -- *console*     0x00  L    1  C  0   Top  empty  0                                                     
:yi errorEditor "hello"yi-linux-x86_64: ~/.cabal/lib/x86_64-linux-ghc-7.10.3/charsetdetect-ae-1.1.0.1-Jetbu8hLY8r8wvkM6U2eCW/libHScharsetdetect-ae-1.1.0.1-Jetbu8hLY8r8wvkM6U2eCW.a: unhandled ELF relocation(RelA) type 42
 -- *console*     0x00  L    1  C  0   Top  empty  0                                                     
error: GhcException "unable to load package `charsetdetect-ae-1.1.0.1'"                                  
```

**Comment by noughtmare** [comment]
This is probably due to someone removing the `charsetdetect-ae` package dependency without testing if the `hint` flag would still work.

**Comment by noughtmare** [comment]
I found out that charsetdetect-ae is used by yi-rope. So that may have something to do with this.

**Comment by jhance** [comment]
That's a reasonable conclusion.

**Comment by noughtmare** [comment]
It seems like this is a problem with `charsetdetect-ae`, because I can load other FFI functions with hint.

**Comment by jhance** [comment]
Have you tried fiddling with hint by hand (in ghci or something) to ensure its hints fault?

**Comment by noughtmare** [comment]
Well, the functions in charsetdetect work in ghc, ghci and with runhaskell, so I think it's an issue with hint.

**Comment by noughtmare** [comment]
I have an update:
If you compile with `ghc-options: -dynamic` the command works.

**Note:** If you compile with dyre you also need to tell dyre to compile with `-dynamic`:
```
$ cd ~/.config/yi
$ touch yi.hs
$ yi --ghc-option=-dynamic
```
**Note2:** If you're using stack you need to launch yi with `stack exec`, because hint needs the yi package to be available.

**Comment by noughtmare** [comment]
This is fixed with ghc 8.0.2 🎉 !

**Comment by noughtmare** [comment]
Now I'm getting errors about the fact that ghc is not thread safe...

**Comment by noughtmare** [comment]
I've made a pull request to fix the thread safe error: #970. After that is merged this issue can be closed.
